xfuser/model_executor/layers/attention_processor.py

```diff
@@ -42,11 +45,62 @@ def is_v100():
     device_name = torch.cuda.get_device_name(torch.cuda.current_device())
     return "V100" in device_name
 
+
 def torch_compile_disable_if_v100(func):
     if is_v100():
         return torch.compiler.disable(func)
     return func
 
+
+def apply_rotary_emb(
```

Comment:
This change result in the following error:
```log
torchrun --nproc_per_node=2 ./examples/flux_example.py --model /cfs/dit/FLUX.1-dev --ulysses_degree 2 --prompt "A snowy mountain" --num_inference_steps 20

[rank0]: Traceback (most recent call last):
[rank0]:   File "~/xDiT/./examples/flux_example.py", line 77, in <module>
[rank0]:     main()
[rank0]:   File "~/xDiT/./examples/flux_example.py", line 35, in main
[rank0]:     pipe.prepare_run(input_config)
[rank0]:   File "~/xDiT/xfuser/model_executor/pipelines/pipeline_flux.py", line 69, in prepare_run
[rank0]:     self.__call__(
[rank0]:   File "~/miniconda3/envs/long_ctx_attn/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
[rank0]:     return func(*args, **kwargs)
[rank0]:   File "~/xDiT/xfuser/model_executor/pipelines/base_pipeline.py", line 181, in wrapper
[rank0]:     return func(*args, **kwargs)
[rank0]:   File "~/xDiT/xfuser/model_executor/pipelines/base_pipeline.py", line 133, in data_parallel_fn
[rank0]:     return func(self, *args, **kwargs)
[rank0]:   File "~/xDiT/xfuser/model_executor/pipelines/base_pipeline.py", line 149, in check_naive_forward_fn
[rank0]:     return func(self, *args, **kwargs)
[rank0]:   File "~/xDiT/xfuser/model_executor/pipelines/pipeline_flux.py", line 297, in __call__
[rank0]:     latents = self._sync_pipeline(
[rank0]:   File "~/xDiT/xfuser/model_executor/pipelines/pipeline_flux.py", line 399, in _sync_pipeline
[rank0]:     latents, encoder_hidden_states = self._backbone_forward(
[rank0]:   File "~/xDiT/xfuser/model_executor/pipelines/pipeline_flux.py", line 484, in _backbone_forward
[rank0]:     noise_pred, encoder_hidden_states = self.transformer(
[rank0]:   File "~/miniconda3/envs/long_ctx_attn/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
[rank0]:     return self._call_impl(*args, **kwargs)
[rank0]:   File "~/miniconda3/envs/long_ctx_attn/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
[rank0]:     return forward_call(*args, **kwargs)
[rank0]:   File "~/xDiT/xfuser/model_executor/models/transformers/transformer_flux.py", line 147, in forward
[rank0]:     encoder_hidden_states, hidden_states = block(
[rank0]:   File "~/miniconda3/envs/long_ctx_attn/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
[rank0]:     return self._call_impl(*args, **kwargs)
[rank0]:   File "~/miniconda3/envs/long_ctx_attn/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
[rank0]:     return forward_call(*args, **kwargs)
[rank0]:   File "~/miniconda3/envs/long_ctx_attn/lib/python3.10/site-packages/diffusers/models/transformers/transformer_flux.py", line 200, in forward
[rank0]:     attn_output, context_attn_output = self.attn(
[rank0]:   File "~/miniconda3/envs/long_ctx_attn/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
[rank0]:     return self._call_impl(*args, **kwargs)
[rank0]:   File "~/miniconda3/envs/long_ctx_attn/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
[rank0]:     return forward_call(*args, **kwargs)
[rank0]:   File "~/xDiT/xfuser/model_executor/layers/attention_processor.py", line 223, in forward
[rank0]:     return self.processor(
[rank0]:   File "~/xDiT/xfuser/model_executor/layers/attention_processor.py", line 705, in __call__
[rank0]:     query = apply_rotary_emb(query, image_rotary_emb)
[rank0]:   File "~/xDiT/xfuser/model_executor/layers/attention_processor.py", line 76, in apply_rotary_emb
[rank0]:     cos, sin = freqs_cis  # [S, D]
[rank0]: ValueError: not enough values to unpack (expected 2, got 1)
```

We can merge this pr again after the problem is fixed.

Reply:
Oh, looks like the latest `diffusers` has changed a lot and make itself incompatible with the flux implementation in `xdit` anymore.
I guess a refactor could be needed.

Reply:
After modifying the attention processor and remove resuing encoder_hidden_states it now works with `diffusers==0.30.2`.
Anyway it is still only for reference for you. I guess the frequent changes in `diffusers` and lack of some automatic code patching mechanism could be challenging to fit. So those changes need to be trated carefully.

https://github.com/xdit-project/xDiT/pull/264